File: 프로그래머스/1/42748. K번째수/K번째수.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int a,b,c;
vector<int> answer;

vector<int> solution(vector<int> array, vector<vector<int>> commands) {
    for (int i = 0; i< commands.size(); i++) {
        a = commands[i][0];
        b = commands[i][1];
        c = commands[i][2];
        
        vector<int> v;
        for(int j = a-1; j < b; j++) {
            v.push_back(array[j]);
        }
        sort(v.begin(), v.end());
        
        answer.push_back(v[c-1]);
    }
    return answer;
}

File: 프로그래머스/1/42748. K번째수/K번째수.java
import java.util.*;

class Solution {
    public int[] solution(int[] array, int[][] commands) {
        List<Integer> answer = new ArrayList<>();
        
        for (int[] arr : commands) {
            int st = arr[0];
            int ed = arr[1];
            int len = arr[2];
            
            List<Integer> temp = new ArrayList<>();
            for (int i = st - 1; i < ed; i++) {
                temp.add(array[i]);
            }
            
            Collections.sort(temp);
            answer.add(temp.get(len-1));
        }
        
        return answer.stream().mapToInt(i->i).toArray();
    }
}